data_structures/disjoint_set/disjoint_set_2.py

```diff
@@ -0,0 +1,91 @@
+n = 9
+
+parent = list(range(n))
+
+rank = [0] * n
+
+
+def union_set(x: int, y: int) -> None:
```

Comment:
The `union_set` function has a logic error. It checks if x is equal to y before performing any operations, but it should be checking if `find_set(x)` is equal to `find_set(y)`. Replace `if x == y:` with `if find_set(x) == find_set(y):`.